homeassistant/helpers/state.py

```diff
@@ -197,10 +197,11 @@ def async_handle_service_calls(coro_list):
         yield from asyncio.wait(execute_tasks, loop=hass.loop)
 
 
-def state_as_number(state):
+def state_as_number(state, mapping=None):
```

Comment:
Seems like that could cause a bunch of code duplication if this was useful for other metric only history components (datadog or graphite for example).  Would it maybe be better to just add it as its own function in state.py instead of piggybacking on state_as_number?

I'm happy to just shove the logic in the statsd component as it solves my use case fine, just figured it might be helpful to make it globally accessible for anyone else.